plugins/flytekit-omegaconf/flytekitplugins/omegaconf/flytekit_patch.py

```diff
@@ -0,0 +1,64 @@
+import dataclasses
+import inspect
+import logging
+from typing import Type, get_args
+
+from flytekit.core.type_engine import TypeEngine, TypeTransformer, is_annotated
+
+logger = logging.getLogger()
+
+
+def iterate_get_transformers(python_type: Type) -> TypeTransformer:
```

Comment:
can you expand on why we need this?

Reply:
@SebS94 What's the reason again we can't directly identify the right transformer, but need to try them out one by one?

Reply:
The main reason is serialisation of objects with multi-inheritance or Union types. The current flytekit implementation will pick the first matching TypeTransformer and fail if it is unable to serialise the type correctly. Instead we want to try out all viable Transformers and only fail if none of them work.

We encounter these cases quite frequently when building (structured) configs with hydra where nodes will accept different subconfig types.

Reply:
Thanks, added this information to the docstring

Reply:
I know this used to be a problem for string enums where the type engine happened to try the string transformer first, found a match, and then picked this one even though from a users perspective the enum transformer would have been a better pick.

I fixed this special case a while ago (see [here](https://github.com/flyteorg/flytekit/blob/94a48ae86d9b6924bd01c106b493fe50a5169ebb/flytekit/core/type_engine.py#L973)).

Are there other cases where this is required or could it be that this logic here predates the fix I mentioned above and might actually not be required anymore now?

Reply:
Resolving as enum transformer was removed, see https://github.com/flyteorg/flytekit/pull/2299#issuecomment-2106344696.